### PR TITLE
Fix for appstore and wipemycal render

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

Fix for when viewing the app page for a single app in app-store, now it should correctly display if it's already installed or not, this is because we now use the same data that we use on installed apps page, we verify for credentials on user.
Also fix for wipe-my-cal, render conditional.

Fixes ...#2537

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Visit any app page, if it's already installed it should display Installed, if not try to install it and return to the app page after finishing the installation and it should display correctly as Installed.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

## Proof
https://www.loom.com/share/dd5834951049434bb3ffa6d1b7756760

### CODE REVIEW

File: packages/app-store/_utils/useAddAppMutation.ts [hunk 1]
```diff
@@ -12,12 +12,14 @@ function useAddAppMutation(type: App["type"], options?: Parameters<typeof useMut
...
     const searchParams = `?state=${stateStr}`;
+
     const res = await fetch(`/api/integrations/${appName}/add` + searchParams);
+
     if (!res.ok) {
       throw new Error("Something went wrong");
     }
-    const json = await res.json();
-    window.location.href = json.url;
+
```
Comment (alannnc) [inline packages/app-store/_utils/useAddAppMutation.ts:21]: I was having some errors while doing res.json() saying the body wasn't parsing correctly.

File: packages/app-store/_utils/useAddAppMutation.ts [hunk 2]
```diff
@@ -12,12 +12,14 @@ function useAddAppMutation(type: App["type"], options?: Parameters<typeof useMut
...
+    window.location.href = res.url;
```
Comment (alannnc) [inline packages/app-store/_utils/useAddAppMutation.ts:22]: In the end we can access url without doing await res.json();

File: packages/app-store/_utils/useAddAppMutation.ts (hunk 1 above)
Comment (zomars) [reply packages/app-store/_utils/useAddAppMutation.ts:21]: Does this breaks for the other apps?

**Review by zomars — APPROVED** [review]

**Comment by zomars** [comment]
Let's just checks why tests are failing.

File: packages/app-store/_utils/useAddAppMutation.ts (hunk 1 above)
Comment (alannnc) [reply packages/app-store/_utils/useAddAppMutation.ts:21]: It didn't break on my end, and it shouldn't broke for other apps.
@joeauyeung Can you help me test this one?

File: apps/web/components/App.tsx [hunk 3]
```diff
@@ -55,6 +57,11 @@ export default function App({
 }) {
   const { t } = useLocale();
 
+  const { isSuccess, isLoading, data } = trpc.useQuery(["viewer.integrations"]);
+  const appCredentials: { credentialIds: number[] | undefined } | undefined = data?.other?.items.find(
```
Comment (joeauyeung) [inline apps/web/components/App.tsx:61]: I noticed this only searches for credentials for anything under "other" and not the other categories. I'm working on a fix for this right now

Comment (alannnc) [reply apps/web/components/App.tsx:61]: Oh yeah, you are right there. We should iterate on all kinds before jumping on to this.

Comment (zomars) [reply apps/web/components/App.tsx:61]: We may want to handle this from the server. Make an endpoint that specifically returns if an app is installed or not.

Comment (alannnc) [reply apps/web/components/App.tsx]: I've updated the code, and now it should use a endpoint

**Review by zomars — APPROVED** [review]
LGTM

File: apps/web/components/App.tsx [hunk 4]
```diff
@@ -60,7 +59,29 @@ export default function App({
...
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        if (result.status === 200) {
+          setInstalledApp(true);
+        }
+      } catch (error) {
+        if (error instanceof Error) {
+          console.log(error.message);
```
Comment (zomars) [inline apps/web/components/App.tsx:79]: Let's try to move this into a trpc query @alannnc @joeauyeung

Comment (zomars) [inline apps/web/components/App.tsx:79]: Let's try to move this into a trpc query @alannnc @joeauyeung

**Pull request merged**